components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -842,10 +842,12 @@ public static Optional<User> getUser(String tenantDomain, String username)
             throws IdentityApplicationManagementException {
 
         User user = null;
+        String userId = null;
         try {
             int tenantID = IdentityTenantUtil.getTenantId(tenantDomain);
-            String userId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserId();
-
+            if (StringUtils.isBlank(username)) {
+                userId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserId();
+            }
```

Comment:
If we set userId, only if username is blank won't it affect to `getUserFromTenant(username, userId, tenantID)` method?

Reply:
This would not effect it since that function uses either the username or the userId to get the user.
https://github.com/wso2/carbon-identity-framework/blob/7e58b6b480885f14670c445aa5c0a8cd5ade22c7/components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java#L890-L903